docs/designers-developers/developers/block-api/block-edit-save.md

```diff
@@ -226,6 +226,13 @@ For most blocks, the return value of `save` should be an [instance of WordPress
 
 _Note:_ While it is possible to return a string value from `save`, it _will be escaped_. If the string includes HTML markup, the markup will be shown on the front of the site verbatim, not as the equivalent HTML node content. If you must return raw HTML from `save`, use `wp.element.RawHTML`. As the name implies, this is prone to [cross-site scripting](https://en.wikipedia.org/wiki/Cross-site_scripting) and therefore is discouraged in favor of a WordPress Element hierarchy whenever possible.
 
+_Note:_ The save function should be a pure function that depends only on the attributes used to invoke it.
+It can not have any side effect or retrieve information from another source, e.g. it is not possible to use the data module inside it `wp.data.select...`.
+If the function uses external information when the external information changes for the same attribute values, the save result may vary, and when that happens, the block will be considered invalid by the editor.
```

Comment:
Some suggestion for phrasing. I assume the intent here is to explain _why_ it can't be done? May also help to direct people to the Validation section, which includes more information about how validation actually works (and to avoid repeated information).

```suggestion
This is because if the external information changes, the block may be flagged as invalid when the post is later edited ([read more about Validation](#validation)).
```